icu4j/main/classes/core/src/com/ibm/icu/impl/units/UnitPreferences.java

```diff
@@ -56,9 +61,18 @@ private static String[] getAllUsages(String usage) {
         return result.toArray(new String[0]);
     }
 
-    public UnitPreference[] getPreferencesFor(String category, String usage, String region) {
+    public UnitPreference[] getPreferencesFor(String category, String usage, ULocale locale) {
+        if (category.equals("temperature")) {
+            String localeUnit = locale.getKeywordValue("measurement-unit");
```

Comment:
Question: when did the "measurement-unit" identifier get added to ICU?

Reply:
I do not know, but it is added to this file: ../icu4c/source/data/misc/keyTypeData.txt

Reply:
That was imported from CLDR, a long alias.

Reply:
That alias should not have been defined in CLDR. It is unnecessary, excessively long, and, as @younies found out, violates ICU locale keyword syntax. We are lucky that Younies found this before CLDR 42 was released, because we can still retract this alias. I submitted CLDR-16022 to hopefully prevent new key/value aliases going forward.

Reply:
Thanks Markus, and I have updated this PR to use "mu"